activation_key_logger_test.go

```diff
@@ -0,0 +1,457 @@
+				"val",
+			},
+			ExpectedResult: false,
+		},
+
+		// Case 4, same as 4 but with different activation keys.
+		{
+			Activations: map[string]interface{}{
+				"foo": "bar",
+				"bar": "foo",
+				"baz": "foo",
+			},
+			KeyVals: []interface{}{
+				"test",
+				3,
+				"key",
+				"val",
+			},
+			ExpectedResult: false,
+		},
+
+		// Case 5, a given activation key matching any keyVals results into true.
+		{
+			Activations: map[string]interface{}{
+				"test": 3,
+			},
+			KeyVals: []interface{}{
+				"test",
+				3,
+				"key",
+				"val",
+			},
+			ExpectedResult: true,
+		},
+
+		// Case 6, same as 6 but with different matching activation keys.
+		{
+			Activations: map[string]interface{}{
+				"test": 3,
+				"key":  "val",
+			},
+			KeyVals: []interface{}{
+				"test",
+				3,
+				"key",
+				"val",
+			},
+			ExpectedResult: true,
+		},
+
+		// Case 7, activation keys matching values of the keyVals result in false,
+		// because we only want to activate on matching keys.
+		{
+			Activations: map[string]interface{}{
+				"val": "key",
+			},
+			KeyVals: []interface{}{
+				"key",
+				"val",
+			},
+			ExpectedResult: false,
+		},
+
+		// Case 8, activation keys matching keys of the keyVals still result in true
+		// even if values match as well.
+		{
+			Activations: map[string]interface{}{
+				"key": "val",
+				"val": "key",
+			},
+			KeyVals: []interface{}{
+				"key",
+				"val",
+				"val",
+				"key",
+			},
+			ExpectedResult: true,
+		},
+
+		// Case 9, activation keys must all match in order to result in true.
+		{
+			Activations: map[string]interface{}{
+				"foo": "val",
+				"bar": "val",
+				"baz": "val",
+			},
+			KeyVals: []interface{}{
+				"foo",
+				"val",
+				"bar",
+				"val",
+				"baz",
+				"val",
+			},
+			ExpectedResult: true,
+		},
+
+		// Case 10, not all activation keys matching results in false.
+		{
+			Activations: map[string]interface{}{
+				"foo": "val",
+				"bar": "val",
+				"baz": "val",
+			},
+			KeyVals: []interface{}{
+				"foo",
+				"val",
+				"bar",
+				"val",
+				"notmatching",
+				"val",
+			},
+			ExpectedResult: false,
+		},
+	}
+
+	for i, tc := range testCases {
+		result, err := shouldActivate(tc.Activations, tc.KeyVals)
+		if err != nil {
+			t.Fatalf("case %d expected %#v got %#v", i, nil, err)
+		}
+
+		if result != tc.ExpectedResult {
+			t.Fatalf("case %d expected %#v got %#v", i, tc.ExpectedResult, result)
+		}
+	}
+}
+
+func Test_ActivationKeyLogger_shouldActivate_level(t *testing.T) {
+	testCases := []struct {
+		Activations    map[string]interface{}
+		KeyVals        []interface{}
+		ExpectedResult bool
+	}{
+		// Case 0, activation keys representing common log levels result in true
+		// when matching.
+		{
+			Activations: map[string]interface{}{
+				"level": "info",
+			},
+			KeyVals: []interface{}{
+				"test",
+				3,
+				"level",
+				"info",
+			},
+			ExpectedResult: true,
+		},
+
+		// Case 1, same as 12 but with a different log level.
```

Comment:
Note that I have to fix the descriptions still. I want to get some feedback before polishing everything to not waste time in case this here is not what we want.